nexus-and-alo/usage-scenarios.md

```diff
@@ -0,0 +1,32 @@
+> Note: Some of these scenarios are forward looking and will not be supported for some time. 
+
+**Contents:**
+
+- [Exposing Temporal application to a non-Temporal consumer (Nexus + ALO)](#exposing-temporal-application-to-a-non-temporal-consumer-nexus--alo)
+- [Exposing Temporal application to a Temporal consumer (Nexus)](#exposing-temporal-application-to-a-temporal-consumer-nexus)
+- [Exposing non-Temporal application with Temporal-esque calling semantics (ALO)](#exposing-non-temporal-application-with-temporal-esque-calling-semantics-alo)
+- [Exposing non-Temporal application via Nexus](#exposing-non-temporal-application-via-nexus)
+
+## Exposing Temporal application to a non-Temporal consumer (Nexus + ALO)
+
+**Problem statement:** One of the most common patterns we see from Temporal users is the need to expose a generic API for internal consumers who are not familiar with Temporal. The most common solution for this problem today is to stand up a separate infrastructure dependency in front of Temporal (such as a load balancer, HTTP server, GRPC Gateway) which defines the formal API & contract for consumers.
+
+**Solution:** Temporal users can now define and create a Nexus which represents the desired shape of their GRPC/HTTP API. Because Nexus gives users full control of how to parse/handle incoming requests, they can choose to expose their Temporal logic without anyone knowing that they used Temporal behind the scenes. If the consumer of their API has adopted the ALO standard, they can also “automagically” expose their Nexus APIs as ALOs. The capabilities of Nexus completely and totally obviate the need for a standalone API in these scenarios. 
```

Comment:
To clarify somewhere in this doc (because it can be hard to see end goals):

Also, in later versions assuming you don't need advanced runtime request customization, high-level abstractions will make this so transparent to users that a Nexus service definition is all that a client will require to make calls and all that an implementer will require to implement.

The end goal would support someone writing a Nexus service definition. That service definition is discoverable (marketplace and/or just an URL on the endpoint). Client can code-gen from that service definition in any language to have type-safe invocations. Implementer can code-gen from that service definition in non-Temporal-specific way to have simple, typed Nexus interface, or in Temporal-specific way to have a workflow (or top-level activity) interface ready to implement. Of course we'd leverage as much existing industry things as we could.